File: SHSegmentedControlTableView/Demo/TestOneTableView.m
//
//  TestOneTableView.m
//  SHSegmentedControlTableView
//
//  Created by angle on 2017/10/19.
//  Copyright © 2017年 angle. All rights reserved.
//

#import "TestOneTableView.h"

@interface TestOneTableView ()<UITableViewDelegate,UITableViewDataSource>

@end

@implementation TestOneTableView

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/
- (void)setLabel:(NSString *)label {
    _label = label;
    [self reloadData];
}
- (void)setNum:(NSInteger)num {
    _num = num;
    [self reloadData];
}
- (instancetype)init {
    if (self = [super init]) {
        self.num = 0;
        self.dataSource = self;
        self.delegate = self;
        self.rowHeight = 45;
        self.tableFooterView = [[UIView alloc] init];
        [self registerClass:UITableViewCell.class forCellReuseIdentifier:@"cell"];
    }
    return self;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.num;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.textLabel.text = [NSString stringWithFormat:@"%@-%ld-%ld",self.label,indexPath.section,indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
    {
        NSLog(@"%ld-%ld",indexPath.section,indexPath.row);
    }

@end

File: Swift/SHSegmentedControlTableView/SHSegmentedControlTableView/TestOneTableView.swift
//
//  TestOneTableView.swift
//  SHSegmentedControlTableView
//
//  Created by angle on 2018/1/30.
//  Copyright © 2018年 angle. All rights reserved.
//

import UIKit

import SHSegmentedControl

class TestOneTableView: SHTableView, UITableViewDataSource, UITableViewDelegate {
    
    open var label:String!
    
    open var num:NSInteger = 0
    
    override init(frame: CGRect, style: UITableViewStyle) {
        super.init(frame: frame, style: style)

        self.dataSource = self
        self.delegate = self
        self.rowHeight = 45
        self.tableFooterView = UIView.init()
        self.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: "cell")
        self.allowsSelection = false
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    open func setNum(num: NSInteger) {
        self.num = num;
        self.reloadData()
    }
    
    open func setLabel(label: String) {
        self.label = label;
        self.reloadData()
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.num
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell:UITableViewCell = tableView.dequeueReusableCell(withIdentifier: "cell")!
        cell.textLabel?.text = self.label + String.init(format: "%ld -- %ld", indexPath.section, indexPath.row)
        return cell
    }
}
